### PR TITLE
Add support for older Emd versions

### PR DESCRIPTION
**Pull request by ercius**
This adds support for EMD v0.1 or possibly v0.2 emd files. 

A file with one emd group is read and returned as an `emdfile.Array`.

A file with more than one emd group is read and all groups are returned as `emdfile.Root`.

Added `pytest`-based fixtures and tests to write and read temporary files.

### CODE REVIEW

File: src/emdfile/read_EMD_v0p1.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        """Checks if a HDF5 item (group or dataset) is a emd v0.1 group. This is used in h5py.visititems."""
+        if isinstance(obj, h5py.Group):
+            if 'emd_group_type' in obj.attrs and obj.attrs['emd_group_type'] == 1:
+                self.emds.append(obj)
+    
+    def _is_EMD_v0p1(self):
+        if self.file_hdl.attrs['version_major'] == 0 and self.file_hdl.attrs['version_minor'] == 1:
+            return True
+        else:
+            return False
```
Comment (bsavitzky) [inline src/emdfile/read_EMD_v0p1.py:52]: I think this method is unused - is this right?  If so, recommend removing

File: test/test_emd01.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        def test_read2(self, _make_data_2):
+            """Read file with 2 data sets"""
+            new_emd = read_EMD_v0p1(_make_data_2)
+            print(new_emd)
+            
+        def test_read3D(self, _make_data_3D):
+            """Read file with 3D data data sets"""
+            new_emd = read_EMD_v0p1(_make_data_3D)
+            print(new_emd)
+
```
Comment (bsavitzky) [inline test/test_emd01.py:151]: recommend adding a test which reads the test data using the `emdfile.read` method

**Review by bsavitzky — COMMENTED** [review]
Looks great, thanks Peter!  I added a couple of small recs.  I'm going to pull the branch locally to run tests now

**Comment by bsavitzky** [comment]
@ercius Ok!  I made a few edits and handled merge conflicts in Array.__repr__.

**Comment by ercius** [comment]
Looks good to merge to me! Thanks for cleaning things up.

**Review by bsavitzky — APPROVED** [review]

**Pull request merged**